domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/CpuPerformanceSnapshotter.kt

```diff
@@ -36,15 +35,29 @@ class CpuPerformanceSnapshotter(
   private val initialIconificationCutOffTimePeriodMillis: Long
 ) {
 
-  private val commandQueue = createCommandQueueActor()
+  private var isSnapshotterInitialized = false
+  private val commandQueue by lazy { createCommandQueueActor() }
 
-  /** Updates the current [AppIconification] in accordance with the app's state changes. */
+  /** Initialises the snapshotter by calling the commandQueue to start recording CPU Usage. */
+  fun initialiseSnapshotter() {
+    commandQueue
```

Comment:
Moved the "also" logic here. 
Actually, we do need the actor variable to exist -- the presence of the lazy variable ensures that all sendCommands (done via private/public methods) is done on the same instance of the actor. I tried replacing it though (and moved the dependency to maintain channel instance on the actor itself) but it didn't entirely work since methods like `sendSwitchIconificationCommand` require the actor to work but were called from outside the actor and hence didn't have access to the channel.

Reply:
Sorry, I meant the actor _instance_ doesn't need to exist at this point (which I think is correctly demonstrated by the latest code passing all tests). The new code seems a lot cleaner both here & when the actor is initialized.